@xen-orchestra/web-core/lib/utils/interactive-table.util.ts

```diff
@@ -0,0 +1,38 @@
+const COLUMN_NAME_SEPARATOR = '__'
+
+export type InteractionId = 'sort-asc' | 'sort-desc' | 'group' | 'filter' | 'hide'
+
+function compare(prev: unknown, next: unknown) {
+  if (typeof prev === 'string' && typeof next === 'string') {
+    return prev.localeCompare(next)
+  } else if (typeof prev === 'number' && typeof next === 'number') {
+    return prev - next
+  } else if (prev instanceof Date && next instanceof Date) {
+    return prev.getTime() - next.getTime()
+  } else {
+    return String(prev).localeCompare(String(next))
+  }
```

Comment:
It comes down to personal preference and coding style, but what do you think of using this suggestion to improve readability?
```suggestion
  if (typeof prev === 'string' && typeof next === 'string') {
    return prev.localeCompare(next)
  }
  if (typeof prev === 'number' && typeof next === 'number') {
    return prev - next
  }
  if (prev instanceof Date && next instanceof Date) {
    return prev.getTime() - next.getTime()
  }
  return String(prev).localeCompare(String(next))
```